flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -2540,20 +2554,58 @@ void ResolveOmpTopLevelParts(
   });
 }
 
-void OmpAttributeVisitor::CheckDataCopyingClause(
-    const parser::Name &name, const Symbol &symbol, Symbol::Flag ompFlag) {
-  const auto *checkSymbol{&symbol};
+static bool IsSymbolInCommonBlock(const Symbol &symbol) {
+  // If there are many symbols in common blocks, going through them all can be
```

Comment:
Would you like this to be a TODO, instead of a comment, depending on how impactful we expect this to be?

Reply:
Thanks for the review!

Comment changed to a TODO.

For now I don't expect the current implementation of `IsSymbolInCommonBlock` to have a big impact on performance, since it's used only when checking symbols in `copyprivate` clauses. But on programs with an unusually high number of symbols in common blocks and `copyprivate` clauses, or if this predicate function starts to be used within other checks, then it could be a problem.